code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -34,94 +34,500 @@
+			else
+				teleport_ready = FALSE
+				if(teleport_x && teleport_y && teleport_z)
+					var/turf/T = locate(teleport_x, teleport_y, teleport_z)
+					to_chat(user, "<span class='notice'>You snap back to your anchor point!</span>")
+					do_teleport(user, T,  asoundin = 'sound/weapons/emitter2.ogg')
+					return 450
+
 
 /obj/item/slime_extract/pyrite
 	name = "pyrite slime extract"
 	icon_state = "pyrite slime extract"
 
+/obj/item/slime_extract/pyrite/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			var/chosen = pick(difflist(subtypesof(/obj/item/toy/crayon),typesof(/obj/item/toy/crayon/spraycan)))
+			var/obj/item/O = new chosen(null)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/chosen = pick(subtypesof(/obj/item/toy/crayon/spraycan))
+			var/obj/item/O = new chosen(null)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 250
+
 /obj/item/slime_extract/cerulean
 	name = "cerulean slime extract"
 	icon_state = "cerulean slime extract"
 
+/obj/item/slime_extract/cerulean/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			user.reagents.add_reagent("salbutamol",15)
+			to_chat(user, "<span class='notice'>You feel like you don't need to breathe!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/turf/open/T = get_turf(user)
+			if(istype(T))
+				T.atmos_spawn_air("o2=11;n2=41;TEMP=293.15")
+				to_chat(user, "<span class='warning'>You activate [src], and fresh air bursts out of your skin!</span>")
+				return 600
+
 /obj/item/slime_extract/sepia
 	name = "sepia slime extract"
 	icon_state = "sepia slime extract"
 
+/obj/item/slime_extract/sepia/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			var/obj/item/device/camera/O = new(null, 1)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			to_chat(user, "<span class='warning'>You feel time slow down...</span>")
+			if(do_after(user, 30, target = user))
+				var/obj/effect/timestop/T = new(get_turf(user), 2, 50, list(user))
+				T.timestop()
```

Comment:
Forgot to mention, timestop() is already called in Initialize()

Reply:
i've gotta update the slime core effect too it seems

Reply:
oh you already fixed that